core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,289 @@
+    ' cannot create collections', fakeAsync(() => {
+    spyOn(userService, 'getUserInfoAsync')
+      .and.returnValue(Promise.resolve(userInfoForNonCollectionCreator));
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    expect(component.userIsLoggedIn).toBe(false);
+    expect(component.canCreateCollections).toBe(false);
+
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+
+    expect(component.userIsLoggedIn).toBe(true);
+    expect(component.canCreateCollections).toBe(false);
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+  }));
+
+  it('should not start a new creation process if another' +
+    ' creation is in progress', () => {
+    component.canCreateCollections = false;
+    component.creationInProgress = false;
+
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+    expect(component.creationInProgress).toBe(true);
+    expect(explorationCreationServiceSpy).toHaveBeenCalledTimes(1);
+
+    component.initCreationProcess();
+    expect(explorationCreationServiceSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should create new exploration if user cannot create collections', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = false;
+    const explorationCreationServiceSpy = spyOn(
+      explorationCreationService, 'createNewExploration');
+
+    component.initCreationProcess();
+
+    expect(explorationCreationServiceSpy).toHaveBeenCalled();
+  });
+
+  it('should redirect user to a new exploration when user clicks' +
+    ' create button and is not on creator dashboard page', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = true;
+    spyOn(urlService, 'getPathname').and.returnValue(
+      'not/creator-dashboard');
+    spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
+      location: {
+        replace: (val: string) => {}
+      }
+    });
+    const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
+
+    component.initCreationProcess();
+
+    expect(replaceSpy).toHaveBeenCalledWith('/creator-dashboard?mode=create');
+  });
+
+  it('should not redirect user but open a create activity modal if user' +
+    ' can create collections and is on creator dashboard page', () => {
+    component.creationInProgress = false;
+    component.canCreateCollections = true;
+    spyOn(urlService, 'getPathname').and.returnValue(
+      '/creator-dashboard');
+    const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+      return <NgbModalRef>({
+        result: Promise.resolve('success')
+      });
+    });
+    const replaceSpy = spyOn(windowRef.nativeWindow.location, 'replace');
+
+    component.initCreationProcess();
+
+    expect(replaceSpy).not.toHaveBeenCalled();
+    expect(modalSpy).toHaveBeenCalled();
```

Comment:
Should there be a parallel check for this in the previous test?

Reply:
Yes, added